### PR TITLE
Update Zephyr 3.7

### PR DESCRIPTION
**Pull request by iuliana-prodan**
Update Zephyr to 3.7.

With this update we need to rename NXP board conf and overlay since in 3.7 Hardware model v2 was introduced (see https://github.com/zephyrproject-rtos/zephyr/commit/8dc3f856229ce083c956aa301c31a23e65bd8cd8)

Also, fix a compile warning - commit 2.

### CODE REVIEW

**Comment by iuliana-prodan** [comment]
@arnopo @wmamills I've compile tested this PR on all targets.
All are compiling ok, except `zynqmp_r5`:
```
west build -p -b zynqmp_r5_override -s examples/zephyr/rpmsg_multi_services
-- west build: making build dir /openAMP_workspace/openamp-system-reference/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /openAMP_workspace/openamp-system-reference/examples/zephyr/rpmsg_multi_services
-- CMake version: 3.21.1
-- Found Python3: /usr/bin/python3.10 (found suitable version "3.10.15", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/user/.cache/zephyr
-- Zephyr version: 3.7.0 (/openAMP_workspace/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
No board named 'zynqmp_r5_override' found.

Please choose one of the following boards:

stm32f401_mini
black_f407zg_pro
neorv32
icev_wireless
serpente
...
```

I don't know what is the "new" board in Zephyr 3.7 for `zynqmp_r5`.

**Comment by arnopo** [comment]
Thanks @iuliana-prodan for the report


> @arnopo @wmamills I've compile tested this PR on all targets. All are compiling ok, except `zynqmp_r5`:
> 
> ```
> west build -p -b zynqmp_r5_override -s examples/zephyr/rpmsg_multi_services
> -- west build: making build dir /openAMP_workspace/openamp-system-reference/build pristine
> -- west build: generating a build system
> Loading Zephyr default modules (Zephyr base).
> -- Application: /openAMP_workspace/openamp-system-reference/examples/zephyr/rpmsg_multi_services
> -- CMake version: 3.21.1
> -- Found Python3: /usr/bin/python3.10 (found suitable version "3.10.15", minimum required is "3.8") found components: Interpreter 
> -- Cache files will be written to: /home/user/.cache/zephyr
> -- Zephyr version: 3.7.0 (/openAMP_workspace/zephyr)
> -- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
> No board named 'zynqmp_r5_override' found.
> 
> Please choose one of the following boards:
> 
> stm32f401_mini
> black_f407zg_pro
> neorv32
> icev_wireless
> serpente
> ...
> ```
> 
> I don't know what is the "new" board in Zephyr 3.7 for `zynqmp_r5

@tnmysh  , any advice?
regarding Zephyr board documentation (https://docs.zephyrproject.org/latest/boards/index.html#soc=zynqmp_rpu) seems that for the zynqmp we should use the `kv260_r5` target

**Comment by tnmysh** [comment]
For zynqmp board name: qemu_cortex_r5.

**Comment by tnmysh** [comment]
Here is dts: https://github.com/zephyrproject-rtos/zephyr/blob/v3.7-branch/boards/qemu/cortex_r5/qemu_cortex_r5.dts

File: examples/zephyr/rpmsg_multi_services/src/main_remote.c [hunk 1]
```diff
@@ -198,7 +198,7 @@ int mailbox_notify(void *priv, uint32_t id)
 
 int platform_init(void)
 {
-	void *rsc_tab_addr;
+	struct fw_resource_table *rsc_tab_addr;
```
Comment (arnopo) [inline examples/zephyr/rpmsg_multi_services/src/main_remote.c:201]: perhaps this local variable is useless;;; you could remove it and directly use `rsc_table `  line 236

```c
	/* declare resource table region */
	rsc_table_get(&rsc_table , &rcs_size);
	metal_io_init(&device->regions[1], rsc_table, (metal_phys_addr_t *)rsc_table, rcs_size, -1, 0, NULL);
```

**Comment by iuliana-prodan** [comment]
> Thanks @iuliana-prodan for testing with zephyr-3.7.
> 
> For zynqmp board name: qemu_cortex_r5.
> 
> cc: @iuliana-prodan, @arnopo

We already have an overlay for `qemu_cortex_r5` in `examples/zephyr/rpmsg_multi_services/boards/` and, with Zephyr 3.7 this samples compiles for `qemu_cortex_r5`.

Therefore, I can remove this file -  [zynqmp_r5_override.dtsi](https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/zephyr/rpmsg_multi_services/boards/zynqmp_r5_override.dtsi)

File: examples/zephyr/rpmsg_multi_services/src/main_remote.c (hunk 1 above)
Comment (iuliana-prodan) [reply examples/zephyr/rpmsg_multi_services/src/main_remote.c:201]: Sure, will do.

**Comment by tnmysh** [comment]
> > Thanks @iuliana-prodan for testing with zephyr-3.7.
> > For zynqmp board name: qemu_cortex_r5.
> > cc: @iuliana-prodan, @arnopo
> 
> We already have an overlay for `qemu_cortex_r5` in `examples/zephyr/rpmsg_multi_services/boards/` and, with Zephyr 3.7 this sample compiles for `qemu_cortex_r5`.
> 
> Therefore, I can remove this file - [zynqmp_r5_override.dtsi](https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/zephyr/rpmsg_multi_services/boards/zynqmp_r5_override.dtsi)

@wmamills do you agree with above suggestion to remove zynqmp_r5_override.dtsi ?

**Comment by wmamills** [comment]
@tnmysh  qemu_cortex_r5 is for QEMU testing and is not guaranteed to work on real zynqmp HW.  (It has 64MB of TCM memory for example when the real HW has only 64KB at that address in some configurations and at most 256KB. )

If kv260_r5 works I would rather use that.  Right now it may not be better but that is the one we should be fixing to match HW. We should not change qemu_cortex_r5 as it is used as the generic R5 target.

CC @iuliana-prodan

**Comment by tnmysh** [comment]
Thanks @wmamills. I agree. @iuliana-prodan we can't remove that override file as it's used for different board. Also, it will be needed for any future board as well.

**Comment by wmamills** [comment]
@tnmysh Looked at this closer this morning.  kv260_r5 is already there and is the only zynqmp board we document in the README.  It is the board I build for the demo repo both for HW and for QEMU.  So I think it is already OK as is.

(I would really prefer to have a zynqmp_generic board and then build something like zynqmp_generic/r5_lockstep or /r5_split{0,1} as the software for the R5 should work on all zynqmp based boards not just the kv260.)

I would prefer to delete the qemu_cortex_r5 for our demos.  We should definitely leave it out of the README.  We used that before we had kv260_r5 but I don't think we need it anymore.  A generic R5 does not have IPC and A53's running Linux etc. and the qemu_cortex_r5 is suppose to represent a generic R5 even if they use the zynqmp QEMU to run it.

@iuliana-prodan can you please make a new PR to add the  imx8mp_evk/mimx8ml8/adsp to the demo's README.  I presume the std Zephyr SDK to OK for this.

**Review by wmamills — APPROVED** [review]
OK as is or also delete qemu_cortex_r5 overlay if Tanmay agrees

**Comment by iuliana-prodan** [comment]
> @iuliana-prodan can you please make a new PR to add the imx8mp_evk/mimx8ml8/adsp to the demo's README. I presume the std Zephyr SDK is OK for this.

Sure, will do that

**Comment by iuliana-prodan** [comment]
Changes since v1:
- addressed Arnaud comment to remove local variable and use global `rsc_table` (https://github.com/OpenAMP/openamp-system-reference/pull/57#discussion_r1809195421)

**Comment by iuliana-prodan** [comment]
> OK as is or also delete qemu_cortex_r5 overlay if Tanmay agrees

@tnmysh what do you think?
Should I delete or not the `qemu_cortex_r5` overlay?

**Comment by tnmysh** [comment]
@wmamills can we keep it until zynqmp_r5 generic is introduced as suggested above ?

**Comment by wmamills** [comment]
@tnmysh What would generic_zynqmpp_r5 do that kv260_r5 does not?  
What does generic_cortex_r5 do that kv260_r5 does not?
If there is a real reason, we can keep it then that is fine, otherwise it only serves to mislead people.

**Comment by iuliana-prodan** [comment]
> Changes since v1:
> 
> * addressed Arnaud comment to remove local variable and use global `rsc_table` ([Update Zephyr 3.7 #57 (comment)](https://github.com/OpenAMP/openamp-system-reference/pull/57#discussion_r1809195421))

Changes since v2:
- added a new commit to remove `qemu_cortex_r5` as suggested by @wmamills.

**Comment by tnmysh** [comment]
> > Changes since v1:
> > 
> > * addressed Arnaud comment to remove local variable and use global `rsc_table` ([Update Zephyr 3.7 #57 (comment)](https://github.com/OpenAMP/openamp-system-reference/pull/57#discussion_r1809195421))
> 
> Changes since v2:
> 
> * added a new commit to remove `qemu_cortex_r5` as suggested by @wmamills.

Thanks much @iuliana-prodan. Along with this qemu_cortex_r5 should be removed from sample.yaml examples/zephyr/rpmsg_multi_services/sample.yaml.

If you prefer, I can send separate change. Let me know.

**Comment by iuliana-prodan** [comment]
> > > Changes since v1:
> > > 
> > > * addressed Arnaud comment to remove local variable and use global `rsc_table` ([Update Zephyr 3.7 #57 (comment)](https://github.com/OpenAMP/openamp-system-reference/pull/57#discussion_r1809195421))
> > 
> > 
> > Changes since v2:
> > 
> > * added a new commit to remove `qemu_cortex_r5` as suggested by @wmamills.
> 
> Thanks much @iuliana-prodan. Along with this qemu_cortex_r5 should be removed from sample.yaml examples/zephyr/rpmsg_multi_services/sample.yaml.
> 
> If you prefer, I can send separate change. Let me know.

Sorry, I've missed that. I will update it in this PR. Thanks!

**Comment by iuliana-prodan** [comment]
> > Changes since v1:
> > 
> > * addressed Arnaud comment to remove local variable and use global `rsc_table` ([Update Zephyr 3.7 #57 (comment)](https://github.com/OpenAMP/openamp-system-reference/pull/57#discussion_r1809195421))
> 
> Changes since v2:
> 
> * added a new commit to remove `qemu_cortex_r5` as suggested by @wmamills.

Changes since v3:
* addressed @tnmysh comment to remove `qemu_cortex_r5` from `sample.yaml` as well.

**Review by arnopo — APPROVED** [review]
LGTM

**Review by tnmysh — APPROVED** [review]
LGTM.

**Pull request merged**